lib/puppet/util/windows/registry.rb

```diff
@@ -65,6 +65,22 @@ def values(key)
       vals
     end
 
+    def values_by_name(key, names, missing = :error)
```

Comment:
I'm thinking of just removing the `missing` part, because I don't think it's needed. I think we should just ignore missing values and let callers deal with it, which would simplify this code a little.

Reply:
Ok so, a missing valuename would effectively return a `nil` in the hash.  Yeah can drop the `missing` bit

Will probably need some code comments to describe the method better (e.g. names is an array of string (not symbols)) but I assume that'll be coming later as this is still WIP.

nit: Should probably change the method name to `values_by_names` as it's an array of names, not a singular.

nit: Personally I'd prefer `value_names` not `names` as names but I'm not that fussed.